packages/core/src/app/billing/Billing.tsx

```diff
@@ -1,100 +1,62 @@
 import { Legend } from '../ui/form';
 
 import BillingForm, { type BillingFormValues } from './BillingForm';
 import getBillingMethodId from './getBillingMethodId';
 
-export interface BillingProps {
+interface BillingProps {
     navigateNextStep(): void;
-    onReady?(): void;
+    onReady(): void;
     onUnhandledError(error: Error): void;
 }
 
-export interface WithCheckoutBillingProps {
-    countries: Country[];
-    countriesWithAutocomplete: string[];
-    customer: Customer;
-    customerMessage: string;
-    googleMapsApiKey: string;
-    isInitializing: boolean;
-    isUpdating: boolean;
-    shouldShowOrderComments: boolean;
-    billingAddress?: Address;
-    methodId?: string;
-    isFloatingLabelEnabled?: boolean;
-    themeV2?: boolean;
-    getFields(countryCode?: string): FormField[];
-    initialize(): Promise<CheckoutSelectors>;
-    updateAddress(address: Partial<Address>): Promise<CheckoutSelectors>;
-    updateCheckout(payload: CheckoutRequestBody): Promise<CheckoutSelectors>;
-}
-
-class Billing extends Component<BillingProps & WithCheckoutBillingProps> {
-    async componentDidMount(): Promise<void> {
-        const { initialize, onReady = noop, onUnhandledError } = this.props;
+const Billing = ({navigateNextStep, onReady, onUnhandledError}:BillingProps): ReactElement => {
+    const { checkoutService, checkoutState } = useCheckout();
+    const { themeV2 }  = useThemeContext();
 
-        try {
-            await initialize();
-            onReady();
-        } catch (error) {
-            if (error instanceof Error) {
-                onUnhandledError(error);
-            }
-        }
-    }
+    const {
+        data: {
+            getCheckout,
+            getConfig,
+            getCart,
+            getCustomer,
+            getBillingAddress,
+            getBillingAddressFields,
+        },
+        statuses: { isLoadingBillingCountries },
+    } = checkoutState;
+    const config = getConfig();
+    const customer = getCustomer();
+    const checkout = getCheckout();
+    const cart = getCart();
 
-    render(): ReactNode {
-        const { updateAddress, isInitializing, themeV2, ...props } = this.props;
-
-        return (
-            <AddressFormSkeleton isLoading={isInitializing}>
-                <div className="checkout-form">
-                    <div className="form-legend-container">
-                        <Legend testId="billing-address-heading" themeV2={themeV2}>
-                            <TranslatedString id="billing.billing_address_heading" />
-                        </Legend>
-                    </div>
-                    <BillingForm
-                        {...props}
-                        onSubmit={this.handleSubmit}
-                        updateAddress={updateAddress}
-                    />
-                </div>
-            </AddressFormSkeleton>
-        );
+    if (!config || !customer || !checkout || !cart) {
+        throw new Error('Unable to access checkout data')
```

Comment:
Should the message be translated, so the end user can understand what is going on? (I understand that this is a rare error)

Reply:
These errors are the errors in console I believe.

Reply:
Throwing an error here is simply for making TS happy.
In theory, as soon as we stop showing the checkout loading skeleton, all these values are defined.
However, we historically allow them to be `undefined`, hence we need this check quite often.